CEXT-4452: OOPE Tax Module installation & API guide (#340)
AdobeDocs/commerce-extensibility: src/pages/starter-kit/checkout/tax-reference.md

---
title: Payment API reference
description: Learn about the available API endpoints for the payment module in the Adobe Commerce checkout starter kit.
keywords:
  - App Builder
  - REST

>   - Extensibility

Reviewer: @kuflower you can define new keywords for this repo here: https://github.com/AdobeDocs/commerce-extensibility/blob/main/.github/linters/metadata.schema.yml

Our general guidance is to only use keywords defined for ExL (if possible), which can be found here: https://git.corp.adobe.com/AdobeDocs/exl-config/blob/master/metadata-values/feature.yml#L317

---

# Tax API reference

The checkout module provides REST and GraphQL APIs to configure out-of-process tax integrations.

## REST

> The raw REST api schema is available [here](/tax.xml).

Reviewer: ```suggestion
The REST API schema is available [here](/tax.xml).
```

| **Route URL**                                   | **Method** | **Description**                                     |
| ----------------------------------------------- | ---------- | --------------------------------------------------- |

> | `/V1/oope_tax_management/tax_integration/:code` | GET        | Retrieve an OOPE tax integration info by its code.. |

Reviewer: ```suggestion
| `/V1/oope_tax_management/tax_integration/:code` | GET        | Retrieve an OOPE tax integration info by its code.
```

| `/V1/oope_tax_management/tax_integration`       | GET        | List all available tax integration info.            |
| `/V1/oope_tax_management/tax_integration`       | POST       | Create or update an OOPE tax integration.           |

### Create or modify a new OOPE tax integration

> The POST `/V1/oope_tax_management/tax_integration/:code` creates an out-of-process tax integration in the Adobe Commerce instance. Note that, currently, only one tax integration is allowed to be active. If you wish to change the active tax integration, you must first deactivate the current one.

Reviewer: ```suggestion
The POST `/V1/oope_tax_management/tax_integration/:code` endpoint creates an out-of-process tax integration in the Adobe Commerce instance. 

<InlineAlert variant="info" slots="text"/>

Only one tax integration can be active at a time. If you want to change the active tax integration, you must first deactivate the current one.
```

**Payload parameters:**

| Parameter | Type    | Required | Description                                                 |
| --------- | ------- | -------- | ----------------------------------------------------------- |
| `code`    | String  | Yes      | Unique identifier for the tax integration.                  |
| `title`   | String  | Yes      | Display name of the tax integration.                        |
| `active`  | Boolean | No       | Status indicating if the tax integration is active.         |
| `stores`  | Array   | No       | List of store codes where the tax integration is available. |

<CodeBlock slots="heading, code" repeat="2" languages="bash, json" />

#### Example request

```bash
curl --request POST \
--url <ADOBE_COMMERCE_API_URL>/V1/oope_tax_management/tax_integration \
--header 'Authorization: Bearer <TOKEN>' \
--header 'Content-Type: application/json' \
--data '{
"tax_integration": {
    "code": "tax-1",
    "title": "Tax Integration 1",
    "active": true,
    "stores": ["store-1", "store-2"]
}
}'
```

> #### Example response

Reviewer: differentiate examples

Author: What do you mean?

Reviewer: So we don't want to have two examples named "Example Response" in a row, so let's change the name to make it clear to the user what the difference between the examples is.

<CodeBlock slots="heading, code" repeat="2" languages="bash, json" />

#### Example request

```bash
curl --request GET \
--url <ADOBE_COMMERCE_API_URL>/V1/oope_tax_management/tax_integration/:code' \
--header 'Authorization: Bearer <TOKEN>' \
--header 'Content-Type: application/json'
```

#### Example response

```json
{
  "success": true,
  "message": {
    "id": 1,
    "code": "tax-1",
    "title": "Tax Integration 1",
    "active": true,
    "stores": ["store-1", "store-2"]
  }
}
```

### List all OOPE tax integrations

The GET `/V1/oope_tax_management/tax_integration` retrieves a list of all out-of-process tax integrations from the Adobe Commerce instance.

<CodeBlock slots="heading, code" repeat="2" languages="bash, json" />

#### Example request

```bash
curl --request GET \
--url <ADOBE_COMMERCE_API_URL>/V1/oope_tax_management/tax_integration' \
--header 'Authorization: Bearer <TOKEN>' \
--header 'Content-Type: application/json'
```

#### Example response

```json
{
  "success": true,
  "message": [
    {
      "id": 1,
      "code": "tax-1",
      "title": "Tax Integration 1",
      "active": true,
      "stores": ["store-1", "store-2"]
    }
  ]
}
```

## GraphQL

> To check taxes applied by the tax integration, we have two GraphQl queries: `cart` and `customerOrders`.

Reviewer: ```suggestion
There are two GraphQL queries to check taxes applied by the tax integration, `cart` and `customerOrders`.
```
